include/nil/blueprint/macros.hpp

```diff
@@ -0,0 +1,46 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2023 Alexander Evgin <aleasims@nil.foundation>
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//---------------------------------------------------------------------------//
+// @file This file defines some helpful macros, used in assigner.
+//---------------------------------------------------------------------------//
+
+#ifndef NIL_BLUEPRINT_MACROS_HPP
```

Comment:
let's keep style `CRYPTO3_ASSIGNER_`, not the best naming, but we have it in another files
Some files have `NIL_BLUEPRINT`, some `CRYPTO3_ASSIGNER_NIL_BLUEPRINT`, i guess this PR is good place for align all. Let's use `CRYPTO3_ASSIGNER_`

Reply:
We have a really messy include guards at the moment actually. E.g. `assigner.hpp` has:

```cpp
#define CRYPTO3_BLUEPRINT_COMPONENT_INSTRUCTION_PARSER_HPP
```

which doesn't makes any sense at all. Some of the files have something mixed like:

```cpp
#define CRYPTO3_ASSIGNER_NIL_BLUEPRINT_ASSERTS_HPP
```

which is also a bit confusing. So it's really hard to say, which style we use in this project.

There are some style guides like [this one from Google](https://google.github.io/styleguide/cppguide.html#The__define_Guard), which proposes ` <PROJECT>_<PATH>_<FILE>_H_`. So it may be:

```cpp
#define ZKLLVM_ASSIGNER_NIL_BLUEPRINT_MACROS_HPP_
```

Using `CRYPTO3` doesn't makes any sense: there is nothing related to `crypto3` here. `assigner` is not a part of `crypto3`.

I suggest we refer to #180 and step by step change all include guards to something sound and defined. That would be a pretty easy changes to make. Why not starting in this PR? How do you feel about google-style?

Reply:
may be `ZKLLVM_ASSIGNER_MACROS_HPP_` everywhere?

Reply:
Fixed this to be in Google-style, aligned with all other files:

- #214